packages/storefront-events-collector/src/alloy.ts

```diff
@@ -106,6 +87,36 @@ const hasConfig = (): boolean => {
     return !!eventForwarding?.aep && !!config.datastreamId && !!config.imsOrgId;
 };
 
+const getCustomIdentityMap = (ecid: string, schema: BeaconSchema): IdentityMap | CustomIdentityMap => {
+    const { context } = window.magentoStorefrontEvents;
+    const eventForwarding = context.getEventForwarding();
+    const config = context.getAEP();
+
+    if (eventForwarding && config?.identityMap) {
+        return config.identityMap;
```

Comment:
Why we don't merge custom identity with at least ECID? we may set it as not primary, but we can guarantee that all events have ECID

Reply:
It was a requirement to replace completely